cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/reader/StreamScanner.java

```diff
@@ -49,30 +49,36 @@
  * <p>
  * Upon return from the next() call the current values of the scanner can be obtained by calling
  * the methods in Rid, getPartitionKey(), getColumnName(), getValue().
+ *
  * @param <Type> type of object returned by rid() method.
  */
 @SuppressWarnings("unused")
 public interface StreamScanner<Type> extends Closeable
 {
     /**
-     * Expose the data/rid to be consumed.
-     * Implementation note: rid should always be updated to the current partition if hasNext returns true.
+     * Exposes the data to be consumed.
+     * <p>Implementation note: The data should always be updated to the current partition if {@code next} returns {@code true}.
+     * Calls to {@code data} should be idempotent and not alter the state of the underlying scanner.
+     * </p>
      *
-     * @return rid
+     * @return data of type {@code Type}
      */
-    Type rid();
+    Type data();
 
     /**
-     * Indicate if there are more data/rid avaiable
+     * Indicate if there are more data available to read.
+     * <p>Implementation note: {@code next} can alter the state of the underlying scanner,
+     * so it should be called once and not called again until the {@code data} is consumed.
+     * </p>
      *
-     * @return true when the rid is available to be consumed;
+     * @return true when the data is available to be consumed;
      * otherwise, return false to indicate the scanner has exhausted
      * @throws IOException
      */
-    boolean hasNext() throws IOException;
+    boolean next() throws IOException;
```

Comment:
The renaming does not make sense to me. 

As the updated doc states, it expects the consumer to always consume the next data. Meaning, it does not expect calling the `next()` method N times to skip N - 1 items. 

So, what we want is still just an `Iterator`. 

It is easier and less error prone by having the data advancing logic coded in the implementation, instead of the "implementation note". For example, the first call to `hasNext` advance to the next position, but the subsequent calls w/o calling `next()` does not advance and the state does not change.